### PR TITLE
[NETBEANS-4062] Fixed issue : import RECORD from other package hint

### PR DESCRIPTION
**Pull request by singh-akhilesh**
https://issues.apache.org/jira/browse/NETBEANS-4062

### CODE REVIEW

File: java/java.source.base/src/org/netbeans/api/java/source/ElementHandle.java [hunk 1]
```diff
@@ -436,7 +436,7 @@ public boolean signatureEquals (@NonNull final T element) {
      * @param kind the {@link ElementKind} of the {@link TypeElement},
      * allowed values are {@link ElementKind#CLASS}, {@link ElementKind#INTERFACE},
      * {@link ElementKind#ENUM} and {@link ElementKind#ANNOTATION_TYPE}.
-     * @param binaryName the class binary name as specified by JLS §13.1
+     * @param binaryName the class binary name as specified by JLS ยง13.1
```
Comment (junichi11) [inline java/java.source.base/src/org/netbeans/api/java/source/ElementHandle.java:439]: I don't think we have to change this.

File: java/java.source.base/src/org/netbeans/api/java/source/ElementHandle.java [hunk 2]
```diff
@@ -634,7 +634,11 @@ public ElementHandle create(ElementKind kind, String... descriptors) {
                     }
                     return new ElementHandle<VariableElement> (kind, descriptors);
                 default:
-                    throw new IllegalArgumentException ();
+                    if(kind.name().equals(TreeShims.RECORD) &&  (descriptors.length == 1)) {
+                        return new ElementHandle<TypeElement> (kind, descriptors);
+                    }
+                    else throw new IllegalArgumentException ();
```
Comment (junichi11) [inline java/java.source.base/src/org/netbeans/api/java/source/ElementHandle.java:640]: Please use `{}` (for `else`) and reformat your changes.

File: java/java.source.base/src/org/netbeans/api/java/source/GeneratorUtilities.java [hunk 3]
```diff
@@ -1142,7 +1143,12 @@ public CompilationUnitTree addImports(CompilationUnitTree cut, Set<? extends Ele
                     el = e.getEnclosingElement();
                     break;
                 default:
-                    assert false : "Illegal element kind: " + e.getKind(); //NOI18N
+                    if (TreeShims.isRecord(e)) {
+                        if (e.getEnclosingElement().getKind() == ElementKind.PACKAGE) {
+                            el = e.getEnclosingElement();
+                        }
+                    } else
+                        assert false : "Illegal element kind: " + e.getKind(); //NOI18N
```
Comment (junichi11) [inline java/java.source.base/src/org/netbeans/api/java/source/GeneratorUtilities.java:1151]: Please use `{}`.

File: java/java.source.base/src/org/netbeans/api/java/source/ElementHandle.java (hunk 1 above)
Comment (singh-akhilesh) [reply java/java.source.base/src/org/netbeans/api/java/source/ElementHandle.java]: comment addressed

File: java/java.source.base/src/org/netbeans/api/java/source/ElementHandle.java (hunk 2 above)
Comment (singh-akhilesh) [reply java/java.source.base/src/org/netbeans/api/java/source/ElementHandle.java]: comment addressed

File: java/java.source.base/src/org/netbeans/api/java/source/GeneratorUtilities.java (hunk 3 above)
Comment (singh-akhilesh) [reply java/java.source.base/src/org/netbeans/api/java/source/GeneratorUtilities.java]: comment addressed

**Review by lahodaj — APPROVED** [review]
Looks reasonable to me.

**Comment by arvindaprameya** [comment]
Approved, thanks for completing

**Pull request merged**